### DISCUSSION TITLE
uniforms-ionic

### DISCUSSION THREAD

**Discussion by adrianschneider94** [vazco/uniforms#955] [Show and tell]
Hi there,

I've created [Ionic](https://ionicframework.com) bindings for uniforms:
https://github.com/adrianschneider94/uniforms-ionic

Please feel free to try it out. I'm open for suggestions, improvements etc. and would be happy to merge the code into the repo.

I've refactored the common structure of the other packages a lot to ease my own development process. But it should work for everybody.

**Comment by radekmie** [comment]
Hi @adrianschneider94! That's great! Now, the usual approach here would be somehow like this: release it by yourself and if it'll get some traction, we could merge it in later (and transfer the rights to the npm package).

However, there is [uniforms-ionic](https://github.com/aerogear/uniforms-ionic). To some extend, I think it's our fault, that there's no information about community-maintained packages at all. It'd include [uniforms-ionic](https://github.com/aerogear/uniforms-ionic) as well as [uniforms-patternfly](https://github.com/aerogear/uniforms-patternfly) (from the same maintainers).

---

Here's a couple of points regarding your repo:
* There are no tests. That's a deal-breaker for merging it.
* There's no ESLint configured. Consider having one, not necessarily the one used in [vazco/uniforms](https://github.com/vazco/uniforms).
* Chage this `Unstyled` to `Ionic` [here](https://github.com/adrianschneider94/uniforms-ionic/blob/3d3037717209f56fc875165d93f37ac381984ae8/src/forms/BaseForm.tsx#L7) in order to provided better experience in React devtools.
* None of the original themes includes a CSS/SCSS file - we do fallback to inline styles and only if it's **really** needed.
* I have very little experience with Ionic, therefore I'm _a little bit_ skeptical towards maintaining it in general.

In the meantime, feel free to reach out for help or share your suggestions!

**Reply by adrianschneider94** [reply]
Wow, I didn‘t expect that 😂

I wrote a first stub before the aerogear package existed and didn‘t do research now when I got back to it — I would surely have stumbled across a prominent display of community packages so including that in the docs would definitely help!

Sure, my repo is not cleaned up yet, that‘s why I reached out to have a discussion platform — and to share the code if it‘s useful for somebody. I‘ll see if the existing uniforms-ionic fits my needs and then decide how to move forward.

Thanks for your suggestions!
